zk-keygen/src/main.rs

```diff
@@ -0,0 +1,298 @@
+use {
+    bip39::{Mnemonic, MnemonicType, Seed},
+    clap::{crate_description, crate_name, Arg, ArgMatches, Command},
+    solana_clap_v3_utils::{
+        input_parsers::STDOUT_OUTFILE_TOKEN,
+        keygen::{
+            check_for_overwrite,
+            mnemonic::{acquire_language, acquire_passphrase_and_message, WORD_COUNT_ARG},
+            no_outfile_arg, KeyGenerationCommonArgs, NO_OUTFILE_ARG,
+        },
+        DisplayError,
+    },
+    solana_sdk::signer::EncodableKey,
+    solana_zk_token_sdk::encryption::{auth_encryption::AeKey, elgamal::ElGamalKeypair},
+    std::error,
+};
+
+fn output_encodable_key<K: EncodableKey>(
+    key: &K,
+    outfile: &str,
+    source: &str,
+) -> Result<(), Box<dyn error::Error>> {
+    if outfile == STDOUT_OUTFILE_TOKEN {
+        let mut stdout = std::io::stdout();
+        key.write(&mut stdout)?;
+    } else {
+        key.write_to_file(outfile)?;
+        println!("Wrote {source} to {outfile}");
+    }
+    Ok(())
+}
+
+fn app(crate_version: &str) -> Command {
+    Command::new(crate_name!())
+        .about(crate_description!())
+        .version(crate_version)
+        .subcommand_required(true)
+        .arg_required_else_help(true)
+        .subcommand(
+            Command::new("new")
+                .about("Generate a new encryption key/keypair file from a random seed phrase and optional BIP39 passphrase")
+                .disable_version_flag(true)
+                .arg(
+                    Arg::new("key_type")
+                        .long("key-type")
+                        .takes_value(true)
+                        .possible_values(["elgamal", "aes128"])
+                        .value_name("TYPE")
+                        .required(true)
+                        .help("The type of encryption key")
+                )
+                .arg(
+                    Arg::new("outfile")
+                        .long("outfile")
+                        .value_name("FILEPATH")
+                        .takes_value(true)
+                        .help("Path to generated file"),
```

Comment:
Yeah, I also use the `-o` flag often and it is used quite a bit in the solana docs cli examples as well. I added it back in for now, but @t-nelson please let me know if you feel strongly against this. I'd be happy to go either way.

Reply:
my opinion here isn't _strong_.  would be nice to add it back hidden, but i don't see any obvious what to only hide the short opt with clap v3

Reply:
Hm, yeah I don't quite see an easy way to hide just the short opt. Since this is not marked for publication just yet, let's keep the short opt for now. There will be a chance to discuss it again before publication.